src/Injector/AbstractInjector.php

```diff
@@ -162,6 +181,155 @@ public function inject($package, $type, IOInterface $io)
+                foreach ($dependencies as $dependency) {
+                    if (! $this->isRegisteredInConfig($dependency, $config)) {
+                        $io->write(sprintf(
+                            '<error>    Dependency %s is not registered in the configuration</error>',
+                            $dependency
+                        ));
+
+                        return true;
+                    }
+                }
+
+                $lastDependency = $this->findLastDependency($dependencies, $config);
+
+                $pattern = sprintf(
+                    $this->injectionPatterns[self::TYPE_DEPENDENCY]['pattern'],
+                    preg_quote($lastDependency, '/')
+                );
+                $replacement = sprintf(
+                    $this->injectionPatterns[self::TYPE_DEPENDENCY]['replacement'],
+                    $package
+                );
+
+                $config = preg_replace($pattern, $replacement, $config);
+                file_put_contents($this->configFile, $config);
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Find which of dependency packages is the last one on the module list.
+     *
+     * @param array $dependencies
+     * @param string $config
+     * @return string
+     */
+    private function findLastDependency(array $dependencies, $config)
+    {
+        if (count($dependencies) == 1) {
+            return reset($dependencies);
+        }
+
+        $longLength = 0;
+        $last = null;
+        foreach ($dependencies as $dependency) {
+            preg_match(sprintf($this->isRegisteredPattern, preg_quote($dependency, '/')), $config, $matches);
+
+            $length = strlen($matches[0]);
+            if ($length > $longLength) {
+                $longLength = $length;
+                $last = $dependency;
+            }
+        }
+
+        return $last;
+    }
+
+    /**
+     * Inject module $package into $config before the first found application module
+     * and return true.
+     * If there is no any enabled application module, this method will return false.
+     *
+     * @param string $package
+     * @param string $config
+     * @param IOInterface $io
+     * @return bool
+     */
+    private function injectBeforeApplicationModules($package, $config, IOInterface $io)
+    {
+        if ($firstApplicationModule = $this->findFirstEnabledApplicationModule($this->applicationModules, $config)) {
+            $pattern = sprintf(
+                $this->injectionPatterns[self::TYPE_BEFORE_APPLICATION]['pattern'],
+                preg_quote($firstApplicationModule, '/')
+            );
+            $replacement = sprintf(
+                $this->injectionPatterns[self::TYPE_BEFORE_APPLICATION]['replacement'],
+                $package
+            );
+
+            $config = preg_replace($pattern, $replacement, $config);
+            file_put_contents($this->configFile, $config);
```

Comment:
This part looks like duplicated multiple times, and probably it could be improved - see lines 215-225 in this file.